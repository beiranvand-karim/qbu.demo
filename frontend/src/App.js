import React, {Component} from 'react';
import {BrowserRouter as Router, Route, Switch} from "react-router-dom";
import Home from "./containers/Home";
import CreateQuestion from "./containers/CreateQuestion";


class App extends Component {
  render() {
    return (
      <>
        <Router>
          <Switch>
            <Route path="/" exact component={Home} />
            <Route path="/createQuestion" component={CreateQuestion} />
            <Route path="/home" component={Home} />
          </Switch>
        </Router>
      </>
    );
  }
}

export default App;
