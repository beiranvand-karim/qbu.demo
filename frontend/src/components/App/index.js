import React, {Component} from 'react'
import {BrowserRouter as Router, Route} from "react-router-dom"
import {Routes} from "../Routes"


class App extends Component {
  render() {
    return (
      <Router>
        <Route component={Routes}/>
      </Router>
    );
  }
}

export default App;
