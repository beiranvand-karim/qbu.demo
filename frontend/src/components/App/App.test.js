import React from 'react'
import ReactDOM from 'react-dom'
import App from './'
import {MemoryRouter} from "react-router-dom"
import renderer from "react-test-renderer"

describe('<App />', () => {

   it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<App/>, div);
      ReactDOM.unmountComponentAtNode(div);
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<MemoryRouter><App/></MemoryRouter>).toJSON();
      expect(tree).toMatchSnapshot()
   });

});

