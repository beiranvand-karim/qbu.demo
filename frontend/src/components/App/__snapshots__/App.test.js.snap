// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should matches the snapshot 1`] = `
Array [
  <h1>
    home page
  </h1>,
  <a
    href="/createQuestion"
    onClick={[Function]}
  >
    create a question
  </a>,
  <br />,
  <a
    href="/signIn"
    onClick={[Function]}
  >
    sign in
  </a>,
  <br />,
  <a
    href="/questionListPage"
    onClick={[Function]}
  >
    question list page
  </a>,
  <br />,
  <a
    href="/signUp"
    onClick={[Function]}
  >
    sign up
  </a>,
]
`;
